src/Carousel.js

```diff
@@ -301,9 +307,18 @@ export default {
       }
     },
     restartTimer() {
-      if (this.timer) {
-        this.timer.restart();
-      }
+      this.$nextTick(() => {
+        if (this.timer) {
+          this.timer.stop();
+        }
+        if (this.$props.autoPlay) {
```

Comment:
I dislike having multiply nested if and elses, We could simplify this to:

```js
if (!this.$props.autoPlay) {
  return;
}

if (this.timer) {
  this.timer.start();
  return;
}

this.initAutoPlay();
```

I think this is easier to follow.

Reply:
Will try to simplify the logic a bit.
I'm trying to ensure that if the autoPlay is disabled  we clear the interval so that it does nut run all the time if there's no need for that. That's why the stop is needed. Please do let me know if this sounds ok to you.

Reply:
I was only talking about the code block inside the second if, the first one to stop the timer is perfectly fine.